src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -43,6 +43,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for saga storage. Select another persistence or disable the sagas feature using endpointConfiguration.DisableFeature<Sagas>()");
             }
 
+            if (!context.Container.HasComponent<ISagaIdGenerator>())
+            {
+                context.Container.RegisterSingleton<ISagaIdGenerator>(new DefaultSagaIdGenerator());
```

Comment:
The containers all check for `if HasComponent(componentType)` before registering. The previous check is redundant.
```